pandas/core/indexes/datetimes.py

```diff
@@ -1129,9 +1129,43 @@ def slice_indexer(self, start=None, end=None, step=None, kind=None):
     # Wrapping DatetimeArray
 
     @property
-    def _eadata(self):
-        return DatetimeArray._simple_new(self._data,
-                                         tz=self.tz, freq=self.freq)
+    def _data(self):
```

Comment:
I don't think that `_data` is a property anywhere else. Just an attribute that's set in `_simple_new`.

Actually... how does this even work? If you don't have a setter (which I don't see) then simple_new should fail.

Reply:
Ah I see you set eadata.

It doesn't really matter since we're removing it soon anyway, but I'd prefer that `_eadata` always reference `_data`. That makes keeps all the index classes consistent that `._data` is the actual array, not a property.

Reply:
> I don't think that _data is a property anywhere else. Just an attribute that's set in _simple_new.

That's right.  In this PR we set _eadata inside _simple_new and make _data a property returning _eadata._data.  The freq/tz passthrough doesn't work with _eadata as a property (as in master), so the only question is whether to _also_ set _data in _simple_new.  I chose to make it a property to prevent any shenanigans where they become untied.

Reply:
I think https://github.com/pandas-dev/pandas/pull/24024/files#diff-26a6d2ca7adfca586aabbb1c9dd8bf36R74 is what we want for eadata & freq (and if we can do it here, instead of that PR, then that's best).